src/modules/bunnymq.js

```diff
@@ -0,0 +1,347 @@
+   * @return {Promise}         checkRpc response
+   */
+  publish(queue, msg, options) {
+    // default options are persistent and durable because we do not want to miss any outgoing message
+    // unless user specify it
+    const settings = Object.assign({ persistent: true, durable: true }, options);
+    let message = Object.assign({}, msg);
+    return this._connection.get()
+    .then((channel) => {
+      this.channel = channel;
+
+      // undefined can't be serialized/buffered :p
+      if (!message) message = null;
+
+      this._connection.config.transport.info('bmq:producer', `[${queue}] > `, msg);
+
+      return this.checkRpc(queue, parsers.out(message, settings), settings);
+    })
+    .catch((err) => {
+      if ([ERRORS.TIMEOUT, ERRORS.BUFFER_FULL].includes(err.message)) {
+        throw err;
+      }
+      // add timeout between retries because we don't want to overflow the CPU
+      this._connection.config.transport.error('bmq:producer', err);
+      return utils.timeoutPromise(this._connection.config.timeout)
+      .then(() => this.publish(queue, message, settings));
+    });
+  }
+}
+
+class Consumer {
+
+  constructor(connection) {
+    this._connection = connection;
+    this.channel = null;
+  }
+
+  set connection(value) {
+    this._connection = value;
+  }
+  /**
+   * Get a function to execute on incoming messages to handle RPC
+   * @param  {any} msg   An amqp.node message object
+   * @param  {string} queue The initial queue on which the handler received the message
+   * @return {function}       a function to use in a chaining on incoming messages
+   */
+  checkRpc(msg, queue) {
+    /**
+     * When message contains a replyTo property, we try to send the answer back
+     * @param  {any} content the received message:
+     * @return {any}          object, string, number... the current received message
+     */
+    return (content) => {
+      if (msg.properties.replyTo) {
+        const options = { correlationId: msg.properties.correlationId, persistent: true, durable: true };
+        this._connection.config.transport.info('bmq:consumer', `[${queue}][${msg.properties.replyTo}] >`, content);
+        this.channel.sendToQueue(msg.properties.replyTo, parsers.out(content, options), options);
+      }
+
+      return msg;
+    };
+  }
+
+  /**
+   * @deprecated use subscribe instead
+   * Create a durable queue on RabbitMQ and consumes messages from it - executing a callback function.
+   * Automaticaly answers with the callback response (can be a Promise)
+   * @param  {string}   queue    The RabbitMQ queue name
+   * @param  {object}   options  (Optional) Options for the queue (durable, persistent, etc.)
+   * @param  {Function} callback Callback function executed when a message is received on the queue name, can return a promise
+   * @return {Promise}           A promise that resolves when connection is established and consumer is ready
+   */
+   /* eslint no-param-reassign: "off" */
```

Comment:
is this still needed, will eslint throw this error even though you are just passing it to another function?